Source/WebCore/html/MediaElementSession.cpp

```diff
@@ -1189,6 +1237,11 @@ std::optional<NowPlayingInfo> MediaElementSession::nowPlayingInfo() const
     auto* page = m_element.document().page();
     bool allowsNowPlayingControlsVisibility = page && !page->isVisibleAndActive();
     bool isPlaying = state() == PlatformMediaSession::Playing;
+#if ENABLE(MEDIA_SESSION)
+    if (isPlaying && isDocumentPlayingSeveralMediaStreams(m_element.document()) && page)
+        isPlaying = !page->mutedState().contains(MediaProducerMutedState::AudioIsMuted);
```

Comment:
Is there any way to refactor this so that it doesn't require querying every element in the Document every time `nowPlayingInfo()` is called?

Reply:
Maybe the "playing several media streams" state could be updated in `Document::updateIsPlayingMedia()`?

Reply:
Done